packages/design-system/ui/src/blocks/data-table/components/data-table-sortable-header-cell.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react"
+import { useSortable } from "@dnd-kit/sortable"
+import { CSS } from "@dnd-kit/utilities"
+import { clx } from "@/utils/clx"
+import { Table } from "@/components/table"
+
+interface DataTableSortableHeaderCellProps extends React.HTMLAttributes<HTMLTableCellElement> {
+  id: string
+  children: React.ReactNode
+  isFirstColumn?: boolean
+}
+
+export const DataTableSortableHeaderCell = React.forwardRef<
+  HTMLTableCellElement,
+  DataTableSortableHeaderCellProps
+>(({ id, children, className, style: propStyle, isFirstColumn, ...props }, ref) => {
+  const {
+    attributes,
+    listeners,
+    setNodeRef,
+    transform,
+    transition,
+    isDragging,
+  } = useSortable({
+    id,
+  })
+
+  // Only apply horizontal transform, ignore vertical movement
+  const transformStyle = transform ? {
+    x: transform.x,
+    y: 0,
+    scaleX: transform.scaleX,
+    scaleY: transform.scaleY,
```

Comment:
suggestion: set these to 1 to prevent weird element stretch when dragged over another column edge